docs/DafnyRef/UserGuide.md

```diff
@@ -805,9 +805,82 @@ If no legal candidate entry point is identified, `dafny` will still produce exec
 they will need to be linked with some other code in the target language that
 provides a `main` entry point.
 
-### 24.9.2. extern declarations
-
-TO BE WRITTEN
+### 24.9.2. `extern` declarations
```

Comment:
That's great to fill in this section. However, this section seems like it could be merged with the duplicate from the attributes sections, except what you wrote is much more documented. Can you please read the attribute section and ensure in attributes that there is a link to this user's guide section, maybe like "for more documentation about the usage of {:extern}, please read the user's guide".
https://github.com/atomb/dafny/blob/extern-doc/docs/DafnyRef/Attributes.md